src/Curation/Curation.router.spec.ts

```diff
@@ -92,230 +122,420 @@ describe('when handling a request', () => {
-        )
+      describe('when updating a collection', () => {
+        beforeEach(() => {
+          service = mockServiceWithAccess(CollectionCuration, true)
+          jest.spyOn(service, 'getLatestById').mockResolvedValueOnce(undefined)
+        })
+
+        it('should reject with curation not found message', async () => {
+          await expect(router.updateCollectionCuration(req)).rejects.toThrow(
+            'Curation does not exist'
+          )
+        })
       })
-    })
 
-    describe('when everything is fine', () => {
-      it('should resolve with the updated curation', async () => {
-        mockHasAccessToCollection.mockResolvedValueOnce(true)
+      describe('when updating an item', () => {
+        beforeEach(() => {
+          service = mockServiceWithAccess(ItemCuration, true)
+          jest.spyOn(service, 'getLatestById').mockResolvedValueOnce(undefined)
+        })
 
-        jest
-          .spyOn(Curation, 'getLatestForCollection')
-          .mockResolvedValueOnce({ id: 'curationId' } as any)
+        it('should reject with curation not found message', async () => {
+          await expect(router.updateItemCuration(req)).rejects.toThrow(
+            'Curation does not exist'
+          )
+        })
+      })
+    })
 
-        const updateSpy = jest
-          .spyOn(Curation, 'update')
-          .mockResolvedValueOnce({} as any)
+    describe('when everything is fine', () => {
+      let req: AuthRequest
 
-        const req = {
+      beforeEach(() => {
+        req = {
           auth: { ethAddress: 'ethAddress' },
-          params: { collectionId: 'collectionId' },
+          params: { id: 'some id' },
           body: {
             curation: {
               status: 'rejected',
             },
           },
         } as any
+      })
 
-        await expect(router.updateCuration(req)).resolves.toStrictEqual({})
+      describe('when updating a collection', () => {
+        beforeEach(() => {
+          service = mockServiceWithAccess(CollectionCuration, true)
+          jest
+            .spyOn(service, 'getLatestById')
+            .mockResolvedValueOnce({ id: 'curationId' } as any)
+        })
 
-        expect(updateSpy).toHaveBeenCalledWith(
-          {
-            id: 'curationId',
-            status: 'rejected',
-            updated_at: expect.any(String),
-          },
-          { id: 'curationId' }
-        )
+        it('should resolve with the updated curation', async () => {
+          const updateSpy = jest
+            .spyOn(CollectionCuration, 'update')
+            .mockResolvedValueOnce({} as any)
+
+          await expect(
+            router.updateCollectionCuration(req)
+          ).resolves.toStrictEqual({})
+
+          expect(updateSpy).toHaveBeenCalledWith(
+            {
+              id: 'curationId',
+              status: 'rejected',
+              updated_at: expect.any(String),
+            },
+            { id: 'curationId' }
+          )
+        })
```

Comment:
We could split the test into two its, one for the return data and the other one for the update expectation.